frontend/src/pages/clusterSettings/ClusterSettings.tsx

```diff
@@ -370,25 +376,33 @@ const ClusterSettings: React.FC = () => {
                     Toleration key for notebook pods:{' '}
                   </InputGroupText>
                   <TextInput
-                    isDisabled={!notebookTolerationSettings?.enabled}
+                    isDisabled={!notebookTolerationSettings.enabled}
                     className="odh-number-input"
                     name="tolerationKey"
                     id="toleration-key-input"
                     type="text"
                     aria-label="Toleration key"
-                    value={notebookTolerationSettings?.key ? notebookTolerationSettings?.key : ''}
-                    onChange={async (value: string) => {
-                      const newNotebookTolerationSettings = {
+                    value={notebookTolerationSettings.key}
+                    placeholder={DEFAULT_TOLERATION_VALUE}
+                    validated={
+                      notebookTolerationSettings.error ? ValidatedOptions.error : undefined
+                    }
+                    onChange={(value: string) => {
+                      const newNotebookTolerationSettings: NotebookTolerationFormSettings = {
                         ...notebookTolerationSettings,
                         key: value,
+                        error: TOLERATION_FORMAT.test(value) ? undefined : TOLERATION_FORMAT_ERROR,
                       };
-                      setNotebookTolerationSettings(
-                        newNotebookTolerationSettings as NotebookTolerationSettings,
-                      );
+                      setNotebookTolerationSettings(newNotebookTolerationSettings);
                     }}
                   />
                 </InputGroup>
                 <HelperText>
+                  {notebookTolerationSettings.error && (
+                    <HelperTextItem hasIcon variant="error">
+                      {notebookTolerationSettings.error}
+                    </HelperTextItem>
+                  )}
```

Comment:
Can you change line 419 to   isDisabled={!isSettingsChanged || notebookTolerationSettings.error !== undefined} That should prevent the user from saving bad values.